docs/erc721a-queries.md

```diff
@@ -0,0 +1,84 @@
+# ERC721AQueries
+
+[`erc721a/contracts/extensions/ERC721AQueries.sol`](https://github.com/chiru-labs/ERC721A/blob/main/contracts/extensions/ERC721AQueries.sol)
+
+ERC721A subclass with convenience query functions.
+
+Inherits:
+
+- [ERC721A](erc721a.md)
+
+## Functions
+
+### explicitOwnershipOf
+
+```solidity
+function explicitOwnershipOf(uint256 tokenId) public view returns (TokenOwnership memory)
+```
+
+Returns the [`TokenOwnership`](erc721a.md#tokenownership) struct at `tokenId` without reverting.
+
+This is useful for on-chain and off-chain querying of ownership data for tokenomics.
+
+The returned struct will contain the following values:
+
+- If the `tokenId` is out of bounds:
+  - `addr` = `address(0)`
+  - `startTimestamp` = `0`
+  - `burned` = `false`
+
+- If the `tokenId` is burned:
+  - `addr` = `<Address of owner before token was burned>`
+  - `startTimestamp` = `<Timestamp when token was burned>`
+  - `burned` = `true`
+
+- Otherwise:
+  - `addr` = `<Address of owner>`
+  - `startTimestamp` = `<Timestamp of start of ownership>`
+  - `burned` = `false`
+
+### explicitOwnershipsOf
+
+```solidity
+function explicitOwnershipsOf(
+    uint256[] memory tokenIds
+) external view returns (TokenOwnership[] memory)
+```
+
+Returns an array of `TokenOwnership` structs at `tokenIds` in order.
+
+See [`explicitOwnershipOf`](#explicitOwnershipOf).
+
+### tokensOfOwner
+
+```solidity
+function tokensOfOwner(address owner) external view returns (uint256[] memory)
+```
+
+Returns an array of token IDs owned by `owner`.
+
+This function scans the ownership mapping and is O(totalSupply) in complexity.  
+It is meant to be called off-chain.
+
+See [`tokensOfOwnerIn`](#tokensOfOwnerIn) for splitting the scan into 
+multiple smaller scans if the collection is large enough to 
+cause an out-of-gas error. 
+
+This function should work fine for 10k PFP collections.
```

Comment:
Thoughts on maybe adding a similar warning for a size that would not work?

Reply:
Something like:

Should be avoided for collections > 30K (arbitrary number, btw), and should NOT be used for dynamic supplies like breeding.

Reply:
I am purposely vague about the limit, because different RPC providers may have different gas limits on view functions.